src/Shell/MigrateFromCoreTranslation.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Shell;
+
+use Cake\Console\Shell;
+use Cake\I18n\I18n;
+use Cake\ORM\TableRegistry;
+use League\Flysystem\Adapter\Local;
+
+/**
+ * MigrateFromCoreTranslation shell command.
+ */
+class MigrateFromCoreTranslation extends Shell
+{
+
+    /**
+     * Main method for migration data from table i18n used with default core Translate behavior config.
+     *
+     * TODO: Make it work w/ customized table names
+     *
+     * @return bool|int Success or error code.
+     */
+    public function main()
+    {
+        $i18nTable = TableRegistry::get('i18n');
+
+        //Make sure default is default
+        i18n::locale(i18n::defaultLocale());
+
+        $allFks = $i18nTable->find();
+        $numberOfFks = $allFks->count();
+        $numberOfFksLength = strlen($numberOfFks);
+
+        $this->out('There is a total of: ' . $numberOfFks . ' translations to handle!', 0);
+        $this->hr(1);
+
+        foreach ($allFks->toArray() as $number => $fk) {
```

Comment:
there's no need to cast to an array, this also means the query can't yield-as-it-goes because you need all results to generate the array.